### ISSUE TITLE
Clarify `is_end_of_document_line` in docs

### ISSUE THREAD

**Issue by TomJGooding** [Textualize/textual#4145]
The documentation for the `DocumentNavigator` says that a location "refers to a location, in document-space (in the raw document)".

I would expect the "end of line" in the example below to be at index 9. But the `is_end_of_document_line` is actually based on the _length_ of the row, meaning it is off-by-one.

Perhaps what is meant by "end of line" just needs clarifying the docs, or this might be an unintended bug?

```python
from textual.app import App, ComposeResult
from textual.widgets import TextArea

TEXT = "0123456789"


class ExampleApp(App):
    def compose(self) -> ComposeResult:
        yield TextArea.code_editor(TEXT)

    def on_mount(self) -> None:
        text_area = self.query_one(TextArea)
        expected_eol = (0, 9)
        assert text_area.navigator.is_end_of_document_line(expected_eol)


if __name__ == "__main__":
    app = ExampleApp()
    app.run()
```

**Comment by TomJGooding** [comment]
Sorry I forgot to mention: I think the docstring might need correcting anyway here:

https://github.com/Textualize/textual/blob/e1749ac36a88ae7a015a5c0fc16923df9b6ee3a9/src/textual/document/_document_navigator.py#L107-L108

**Comment by darrenburns** [comment]
It's not off by one - I should maybe clarify in the docs though. The end of the line is the maximum position in a line the cursor can rest on. Since you need to be able to move the cursor **_beyond_** the end of the text (otherwise inserting new text would be impossible), the "end of document line" is `len(document_line)`. It's not "index of the last character".

**Comment by TomJGooding** [comment]
Thanks Darren for the explanation, I think clarifying this in the docs is a good idea.

**Comment by TomJGooding** [comment]
Closed as completed by #4151 (thanks Darren!)
